lib/assertions.js

```diff
@@ -1403,6 +1403,13 @@ module.exports = function (expect) {
         }
     });
 
+    expect.addAssertion([
+        '<array-like> [when] sorted <assertion?>',
+        '<array-like> [when] sorted by <function> <assertion?>'
+    ], function (expect, subject, compareFunction) {
+        return expect.shift(subject.sort(compareFunction));
```

Comment:
-1 to mangling the subject. I suggest: `[].concat(subject).sort(compareFunction)`